client/src/modules/Teams/components/Member.jsx

```diff
@@ -36,10 +36,24 @@ class Member extends Component {
       <Gallery data={member} key={member.user.id} />
     ));
   }
+
+  checkTeamLead() {
+    const { members } = this.props;
+    const currentUserId = localStorage.getItem('userId');
+    let status = false;
+    members.forEach(member => {
+      if (member.role === 'lead' && member.userId === currentUserId) {
+        status = true;
+      }
+    });
```

Comment:
Using `forEach` here means every member will be checked. So, if you have 100 members and the first one checked is a team lead, it still checks the remaining 99. Why not a normal `for` loop, and `break` from the loop when `status` is true.
Or you could do this:
```javascript
const lead = members.find(member => (member.role === 'lead' && member.userId === currentUserId));

return !!(lead);
```

From the above, if returns the first member for which the condition `(member.role === 'lead' && member.userId === currentUserId)` is true (so it doesn't have to keep searching after the lead is found).

If the condition is never true then `lead` will be `undefined`.

`!!(lead)` will be `true` if `lead` is not `undefined`, and `false` otherwise.

Reply:
Thanks for the feedback, I have changed the implementation